src/world.rs

```diff
@@ -107,6 +109,8 @@ impl World {
     /// Send `message` from `src` to `dst`. Delivery is asynchronous and not
     /// guaranteed.
     pub(crate) fn send_message(&mut self, src: SocketAddr, dst: SocketAddr, message: Protocol) {
+        trace!("Send {} {} {}", src, dst, message);
```

Comment:
small nit: i'd rephrase this as:

```rust
trace!(?src, ?dst, ?message, "sending message");
```

this allows those values to be emitted as structured key/value pairs.